### ISSUE TITLE
#allowbreakTargetOnly

### ISSUE THREAD

**Issue by SeasonalFerret** [cabaletta/baritone#776]
## Describe your suggestion
A setting for baritone to only be allowed to break the block it's set to via #mine, and no other block.

## Context
Hypixel has this new skyblock thing, and in the hub is this mine where you can mine ores. It's scattered out a large cave system in random places.
If you break an iron block, it turns to stone. Then to cobblestone, then to bedrock.

Baritone has the hardest time ever trying to navigate this place because it tries to break blocks in it's navigation, sometimes getting itself stuck in a repeated cycle of trying different ways to get around the bedrock that keeps forming. 

Best way to handle this is to just have baritone go up to a block, break that block, then be done with it. No mining of any other block because it'll be assumed that every other block is indestructible. 

## Final checklist
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by CheatTriggers** [comment]
Youre using baritone on hypixel lmao

**Comment by SylveonBottle** [comment]
> Youre using baritone on hypixel lmao

😤

**Comment by usereewrjrjeejr** [comment]
Crine

**Comment by PatrickStar420** [comment]
blocksToAvoidBreaking

**Comment by SylveonBottle** [comment]
> blocksToAvoidBreaking

Because it's more efficient to set every single block as "avoidbreaking" except for the one single one you want to mine, then block that one and unblock the next block you want to mind instead of just using a single command to do all of that for you

**Comment by PatrickStar420** [comment]
> > blocksToAvoidBreaking
> 
> Because it's more efficient to set every single block as "avoidbreaking" except for the one single one you want to mine, then block that one and unblock the next block you want to mind instead of just using a single command to do all of that for you

well for your use you would only need to add one or two blocks for it to work most of the time or you could switch around with blockBreakAdditionalPenalty but I actually agree with you that an inverted version of blocksToAvoidBreaking would be very nice

**Comment by Graicc** [comment]
Did you try legitMine?

**Comment by SylveonBottle** [comment]
Yes, but when climbing up mountains occasionally legitmine breaks blocks to
make stairs

**Comment by voxico** [comment]
This would be sooooooooooooooooooooooo nice for hypixel xD I wish they could add it!

**Comment by ZacSharp** [comment]
We now have `allowBreakAnyway`. It's not automatic but it does the job.

> This would be sooooooooooooooooooooooo nice for hypixel

I hope the anticheat got you.

**Comment by SeasonalFerret** [comment]
Just to confirm my understanding: if I set allowBreak to false and add the specific block types I want to mine (e.g., iron ore) to allowBreakAnyway, Baritone should only break those designated blocks during mining and avoid breaking any blocks while pathing, correct?

**Comment by ZacSharp** [comment]
Yes, that's the way it works.
